src/table/pagination-table/components/head/TableHeadWrapper.tsx

```diff
@@ -70,7 +75,7 @@ const TableHeadWrapper = () => {
           return (
             <StyledHeadCell
               headerStyle={styling.head}
-              style={widthStyle} // add by style to reduce number of classes created by mui
+              style={{ ...widthStyle, ...(isNewHeadCellMenuEnabled ? { cursor: "pointer" } : {}) }} // add by style to reduce number of classes created by mui
```

Comment:
the width style is added inline since it update frequently. I think you can pass the flag prop to the styled component and resolve it there